sql/src/main/java/io/crate/jobs/JobExecutionContext.java

```diff
@@ -182,22 +195,20 @@ public long kill() {
                     numKilled++;
                 }
             }
-            killFuture.set(null);
-        } else {
-            try {
-                killFuture.get();
-            } catch (Throwable e) {
-                LOGGER.warn("Error while waiting for already running kill {}", e);
-                return numKilled;
-            }
+        }
+        try {
+            chainedFuture.get();
+            assert subContexts.values().size() == 0: "unexpected subcontexts there: " +  subContexts.values().size();
+        } catch (Exception e) {
+            Throwables.propagate(e);
```

Comment:
I don't think we handle failure cases in the places where we call kill (e.g. it might break the Reaper in the JobContextService)